tests/catch/unit/memory/hipMallocPitch.cc

```diff
@@ -0,0 +1,296 @@
+      for (size_t depth = 0; depth < extent.depth; ++depth) {
+        char* reinterpretedPtr = reinterpret_cast<char*>(hostPtr.get());
+        size_t index = (pitch * extent.height * depth) + (pitch * height) + width;
+        if (*(reinterpretedPtr + index) != 1) {
+          mismatch = true;
+        }
+      }
+    }
+  }
+  REQUIRE(!mismatch);
+}
+
+class ExtentGenerator {
+ public:
+  static constexpr size_t totalRandomValues{20};
+  static constexpr size_t seed{1337};
+
+  std::uniform_int_distribution<size_t> width_distribution{1, 1024};
+  std::uniform_int_distribution<size_t> height_distribution{1, 100};
+  std::uniform_int_distribution<size_t> depth_distribution{1, 100};
+
+  std::vector<hipExtent> extents2D{};
+  std::vector<hipExtent> extents3D{};
+
+  static ExtentGenerator& getInstance() {
+    static ExtentGenerator instance;
+    return instance;
+  }
+
+ private:
+  ExtentGenerator() {
+    std::mt19937 randomGenerator{seed};
+    extents3D = std::vector<hipExtent>{hipExtent{0, 0, 0}, hipExtent{1, 0, 0}, hipExtent{0, 1, 0},
+                                       hipExtent{0, 0, 1}};
+
+    for (size_t i = 0; i < totalRandomValues; ++i) {
+      extents3D.push_back(hipExtent{width_distribution(randomGenerator),
+                                    height_distribution(randomGenerator),
+                                    depth_distribution(randomGenerator)});
+    }
+
+    extents2D = std::vector<hipExtent>{hipExtent{0, 0, 1}, hipExtent{1, 0, 1}, hipExtent{0, 1, 1}};
+
+    for (size_t i = 0; i < totalRandomValues; ++i) {
+      extents2D.push_back(
+          hipExtent{width_distribution(randomGenerator), height_distribution(randomGenerator), 1});
+    }
+  }
+};
+
+enum class AllocationApi { hipMalloc3D, hipMallocPitch, hipMemAllocPitch };
+
+hipExtent generateExtent(AllocationApi api) {
+  hipExtent extent;
+  if (api == AllocationApi::hipMalloc3D) {
+    auto& extents3D = ExtentGenerator::getInstance().extents3D;
+    extent = GENERATE_REF(from_range(extents3D.begin(), extents3D.end()));
+  } else {
+    auto& extents2D = ExtentGenerator::getInstance().extents2D;
+    extent = GENERATE_REF(from_range(extents2D.begin(), extents2D.end()));
+  }
+
+  return extent;
+}
+
+
+TEST_CASE("Unit_hipMalloc3D_ValidatePitch") {
+  hipPitchedPtr hipPitchedPtr;
+  hipExtent validExtent{generateExtent(AllocationApi::hipMalloc3D)};
+
+  MemoryInfo memBeforeAllocation{createMemoryInfo()};
+  HIP_CHECK(hipMalloc3D(&hipPitchedPtr, validExtent));
+  validateMemory(hipPitchedPtr.ptr, validExtent, hipPitchedPtr.pitch, memBeforeAllocation);
+  HIP_CHECK(hipFree(hipPitchedPtr.ptr));
+}
+
+TEST_CASE("Unit_hipMemAllocPitch_ValidatePitch") {
+  size_t pitch;
+  hipDeviceptr_t ptr;
+  hipExtent validExtent{generateExtent(AllocationApi::hipMemAllocPitch)};
+  MemoryInfo memBeforeAllocation{createMemoryInfo()};
+  unsigned int elementSizeBytes = GENERATE(4, 8, 16);
+
+#if HT_NVIDIA /* EXSWCPHIPT-46 */
```

Comment:
Actually we have a fix internally coming for this.

Reply:
I can update this PR once the fix is on github. 
Or feel free to merge it in its current state (it will not fail) and I can then create a new PR to remove the ifdefs once the fix is merged to hipamd.

Reply:
I believe https://github.com/rocm-developer-tools/hipamd/commit/ffafa224d583ebf4acb4a0a36a6606b05ccab5e9 addresses these scenarios. PR can be updated now.

Reply:
I updated the PR to update all the tests that have been fixed. Unfortunately, I'm unable to test the change because the most recent hipamd build is not building. 

There is also a test that already exists in the Repo but is failing on Nvidia platforms: "Max size_t all dimensions"